### DISCUSSION TITLE
Validate only one path

### DISCUSSION THREAD

**Discussion by samislam** [Automattic/mongoose#12004] [Q&A] [answered]
I have the following long schema:

```js
const mySchema = new mongoose.Schema({
  // some stuff, firstName, lastName ... etc
  password: {
    type: String,
    minLength: 8,
    maxLength: 120,
  }
})
```
And I am inside one of the routes, I want to run validation on the password filed only. Is there something that I can do in Mongoose such as:

```
mySchema.fields.password.validate("123") // Error: password is less than 8 characters!
```

Is there something like that?

**Comment by vkarpov15** [comment, marked as answer]
`mySchema.path('password').doValidate(value, fn)`. Or you can do `doc.validate(['password'])`.

**Reply by samislam** [reply]
can you give the documentation link for `doValidate()`? I couldn't find it, I found `.validate()`  However, it seems like it's exported publicly, I can call it.

**Reply by samislam** [reply]
What are the arguments for this function?  `(value, fn)`, what is fn? is it possible to use the same validations in the schema? the minLength, maxLength, and the others?

**Reply by vkarpov15** [reply]
`fn` is the callback. `doValidate()` is async, but doesn't support promises currently.

**Reply by vkarpov15** [reply]
```javascript
const mySchema = new Schema({
  password: {
    type: String,
    validate: v => v.length >= 6
  }
});

mySchema.path('password').doValidate('abcde', function(err) {
  console.log(err); // ValidatorError
});
```

Does that help?

**Reply by samislam** [reply]
yes, I have implemented this in my code, thank you.
